index.js

```diff
@@ -1,80 +1,113 @@
+checkUser(PROFILE_ID)
+    .then(user => {
+        if (user) {
+            let previousFollowerIds = user.followers;
+            return getAccessToken()
+                .then(accessToken => {
+                    console.log(`Got access token for check difference`, accessToken)
+                    return requestFollowers(accessToken, PROFILE_ID)
+                })
+                .then(currentFollowers => {
+                    fetchFollowers(previousFollowerIds)
+                        .then(previousFollowers => {
+                            let currentFollowerNames = currentFollowers.map(currentFollower => currentFollower.name);
+                            let previousFollowerNames = previousFollowers.map(previousFollower => previousFollower.name);
+                            let newFollowers = currentFollowerNames.filter(x => !previousFollowerNames.includes(x));
+                            let usersToLeaveFollowing = previousFollowerNames.filter(x => !currentFollowerNames.includes(x));
+                            let followersDiff = new FollowersDiff(newFollowers, usersToLeaveFollowing);
+                            console.log("Difference:", followersDiff)
+                        })
+                })
+        }
+    });
 
-  })
+function checkUser(userId) {
+    console.log('Checking User')
 
-function createUser(userId) {
-  console.log('In create User')
+    return User.findOne({id: userId})
+        .then(user => {
+            if (user) {
+                console.log('User exists!')
+                Follower.findById(user.followers[0]._id)
+                    .then(follower => {
+                        console.log(follower);
+                    })
+                return user;
+            } else {
+                console.log('Creating new user')
+                createUser(userId)
+            }
+        })
+}
 
-  return User.findOne({ id: userId })
-    .then(user => {
-      if (user) {
-        console.log('User exists!')
-        // throw new Error('User Exists!')
-        return user;
-      } else {
-        console.log('Creating user')
-        let newUser = new User({
-          id: userId,
-          followers: []
+function createUser(userId) {
+    getAccessToken()
+        .then(accessToken => {
+            console.log(`Got access token`, accessToken)
+            return requestFollowers(accessToken, PROFILE_ID)
+        }).then(followers => {
+        return saveFollowers(followers)
+    }).then(followers => {
+        return new User({
+            id: userId,
+            followers: followers
         })
-        return newUser.save()
-      }
-    })
+    }).then(user => {
+        user.save()
+    });
+}
+
+async function getAccessToken() {
+    const {data} = await get(TOKEN_URL)
+    return data.accessToken;
+}
+
+function fetchFollowers(previousFollowers) {
+    let promises = [];
+    for (follower of previousFollowers) {
+        promises.push(Follower.findById(follower._id))
```

Comment:
Need to fetch followers of user one by one but parallel. Maybe we can improve this logic